packages/astro/src/content/types-generator.ts

```diff
@@ -419,6 +423,29 @@ async function writeContentFiles({
 				for (const entryKey of Object.keys(collection.entries).sort()) {
 					const dataType = collectionConfig?.schema ? `InferEntrySchema<${collectionKey}>` : 'any';
 					dataTypesStr += `${entryKey}: {\n	id: ${entryKey};\n  collection: ${collectionKey};\n  data: ${dataType}\n};\n`;
+					if(settings.config.experimental.contentCollectionJSONSchema) {
+						if (collectionConfig) {
+							// writing inside loop makes sense for the json schema, since we want to have a file per collection
+							const collectionSchemasDir = new URL('./schemas/collections/', contentPaths.cacheDir);
```

Comment:
There's no need to create the same `URL` at every instance of the loop. This variable can be created once before the loop.